File: DevExpress-Demo/DevExpress.SalesDemo.Wpf/CS/DevExpress.SalesDemo.Wpf/ViewModel/SalesSummaryViewModel/SalesSummaryViewModel.cs
using DevExpress.Mvvm.POCO;
using DevExpress.SalesDemo.Model;
using System;

namespace DevExpress.SalesDemo.Wpf.ViewModel {
    public class SalesSummaryViewModel : DataViewModel {
        public static SalesSummaryViewModel Create() {
            return ViewModelSource.Create(() => new SalesSummaryViewModel());
        }
        public SalesSummaryViewModel() {
            SalesBySectorViewModel = SalesBySectorViewModel.Create();
            ThisYearAnnualSalesPerformanceViewModel = AnnualSalesPerformanceViewModel.Create(DateTime.Now);
            LastYearAnnualSalesPerformanceViewModel = AnnualSalesPerformanceViewModel.Create(DateTime.Now.AddYears(-1));
            SalesForecastViewModel = SalesForecastViewModel.Create();
            SalesForecastHeader = string.Format("SALES FORECAST ({0})", DateTimeUtils.GetCurrentYear());
        }

        public virtual SalesBySectorViewModel SalesBySectorViewModel { get; protected set; }
        public virtual AnnualSalesPerformanceViewModel ThisYearAnnualSalesPerformanceViewModel { get; protected set; }
        public virtual AnnualSalesPerformanceViewModel LastYearAnnualSalesPerformanceViewModel { get; protected set; }
        public virtual SalesForecastViewModel SalesForecastViewModel { get; protected set; }
        public virtual string SalesForecastHeader { get; protected set; }
    }
}

File: DevExpress-Demo/DevExpress.SalesDemo.Wpf/VB/DevExpress.SalesDemo.Wpf/ViewModel/SalesSummaryViewModel/SalesSummaryViewModel.vb
Imports DevExpress.Mvvm.POCO
Imports DevExpress.SalesDemo.Model

Namespace DevExpress.SalesDemo.Wpf.ViewModel

    Public Class SalesSummaryViewModel
        Inherits DataViewModel

        Public Shared Function Create() As SalesSummaryViewModel
            Return ViewModelSource.Create(Function() New SalesSummaryViewModel())
        End Function

        Public Sub New()
            SalesBySectorViewModel = SalesBySectorViewModel.Create()
            ThisYearAnnualSalesPerformanceViewModel = AnnualSalesPerformanceViewModel.Create(Date.Now)
            LastYearAnnualSalesPerformanceViewModel = AnnualSalesPerformanceViewModel.Create(Date.Now.AddYears(-1))
            SalesForecastViewModel = SalesForecastViewModel.Create()
            SalesForecastHeader = String.Format("SALES FORECAST ({0})", DateTimeUtils.GetCurrentYear())
        End Sub

        Public Overridable Property SalesBySectorViewModel As SalesBySectorViewModel

        Public Overridable Property ThisYearAnnualSalesPerformanceViewModel As AnnualSalesPerformanceViewModel

        Public Overridable Property LastYearAnnualSalesPerformanceViewModel As AnnualSalesPerformanceViewModel

        Public Overridable Property SalesForecastViewModel As SalesForecastViewModel

        Public Overridable Property SalesForecastHeader As String
    End Class
End Namespace
